text/3848-asm-const-ptr.md

```diff
@@ -0,0 +1,922 @@
+- Feature Name: `asm_const_ptr`
+- Start Date: 2025-07-09
+- RFC PR: [rust-lang/rfcs#3848](https://github.com/rust-lang/rfcs/pull/3848)
+- Rust Issue: [rust-lang/rust#128464](https://github.com/rust-lang/rust/issues/128464)
+
+# Summary
+[summary]: #summary
+
+The `const` operand to `asm!` and `global_asm!` currently only accepts
+integers. Change it to also accept pointer values. The value must be computed
+during const evaluation. The operand expands to the name of the symbol that the
+pointer references, plus an integer offset when necessary.
+
+# Motivation
+[motivation]: #motivation
+
+Right now, the only way to reference a global symbol from inline asm is to use
+the `sym` operand type.
+```rs
+use std::arch::asm;
+
+static MY_GLOBAL: i32 = 10;
+
+fn main() {
+    let mut addr: *const i32;
+    unsafe {
+        asm!(
+            "lea {1}(%rip), {0}",
+            out(reg) addr,
+            sym MY_GLOBAL,
+            options(att_syntax)
```

Comment:
Is there any technical reason using non-default `att_syntax` in most (but not all) examples? Maybe we should mention that?

The only issue I know is that local symbol address immediate (`OFFSET 2f`) cannot be parsed correctly by LLVM (https://github.com/rust-lang/rust/issues/79874), but all examples here use global symbols thus do not suffer from that.

Reply:
if `mov {1}, offset {0} - .` does not work I think there should be an issue for the LLVM assembler 😅

Reply:
You're welcome [to try for yourself](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=d7a55791e7ce6ef72b178e7950b868c2).